app/json_convert/src/main.cpp

```diff
@@ -0,0 +1,188 @@
+        if (!strcmp(argv[i], "-f")) {
+            if (i + 1 > argc) {
+                logger.log_message(
+                    "Must specify a path to the config file after using the -f "
+                    "option");
+                printf(
+                    "Must specify a path to the config file after using the -f "
+                    "option\n");
+                return -1;
+            } else {
+                config_file = argv[++i];
+            }
+        } else {
+            std::string msg = "Invalid argument: ";
+            msg += argv[i];
+            logger.log_message(msg.c_str());
+            printf("Invalid argument: %s\n", argv[i]);
+            return -1;
+        }
+    }
+
+    VCM* vcm;
+
+    if (config_file == "") {
+        vcm = new VCM();
+    } else {
+        vcm = new VCM(config_file);
+    }
+
+    if (FAILURE == vcm->init()) err_handle("VCM failed to initialize", &logger);
+    if (FAILURE == tlm.init()) err_handle("Telemetry Viewer failed to initialize", &logger);
+
+    tlm.add_all();
+    tlm.set_update_mode(TelemetryViewer::BLOCKING_UPDATE);
+
+    unsigned int max_length = 0;
+    size_t max_size = 0;
+    int fields = 0;
+
+    // Sets the max sizes
+    for (std::string it : vcm->measurements) {
+        if (it.length() > max_length) {
+            max_length = it.length();
+        }
+
+        measurement_info_t* info = vcm->get_info(it);
+        // assuming info isn't NULL since it's in the vcm list
+        if (info->size > max_size) {
+            max_size = info->size;
+        }
+        fields++;
+    }
+
+    max_size += (fields * 4) + 2; // Extra characters for JSON formatting
+
+    // Setup UDP server
+    int sockfd;
+    char buffer[max_size];
+    struct sockaddr_in server_addr;
+    struct sockaddr_in client_addr;
+    
+    if ((sockfd = socket(AF_INET, SOCK_DGRAM, 0)) < 0) {
+        err_handle("Socket File Descriptor initialization failed", &logger);
+    }
+
+    logger.log_message("JSON_Convert: Socket setup successful");
+
+
+    bzero(&server_addr, sizeof(server_addr));
+    bzero(&client_addr, sizeof(client_addr));
+
+    server_addr.sin_family = AF_INET;
+    server_addr.sin_addr.s_addr = INADDR_ANY;
+    server_addr.sin_port = htons(SERVER_PORT);
+
+    // client_addr.sin_addr.s_addr = inet_addr(SERVER_IP);
+    // client_addr.sin_port = htons(SERVER_PORT);
+
+    int optval = 1;
+    setsockopt(sockfd, SOL_SOCKET, SO_REUSEADDR, (const void *) &optval, sizeof(int));
+
+    if ((bind(sockfd, (struct sockaddr*) &server_addr, sizeof(server_addr))) < 0) err_handle("Failed to bind", &logger);
+    
+    while (1) {
+        if (killed) {
+            logger.log_message("JSON_Convert: Unaliving process");
+            exit(0);
+        }   
+
+        // Receive data from client to know where to send
+        socklen_t client_len = sizeof(client_addr);
+        int bytes_received = recvfrom(sockfd, buffer, max_size, 0, (struct sockaddr*) &client_addr, &client_len);
```

Comment:
you're still receiving data from somewhere and using that address to send to, I think you want this app to always transmit and not block on receiving. I imagine anything that wants to receive JSON isnt going to be sending any packets to you?